### PR TITLE
Created all required queries

### PR DESCRIPTION
**Pull request by Avyalikov**

### CODE REVIEW

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        }
+
+        [Category("Homework")]
+        [Title("Querie_6")]
+        [Description("This querie return all customers with non-num postcode or without region or without operator code.")]
+
+        public void Querie6()
+        {
+            var customers = dataSource.Customers
+                .Where(a => !SqlMethods.Like(a.PostalCode, "%[^0-9]%") || a.Region == null || !a.Phone.StartsWith("("))
```
Comment (hobowithshotgun) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: `!SqlMethods.Like(a.PostalCode, "%[^0-9]%")` double 'not' here: ! and ^ do the same. fix this one please

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+			{
+				ObjectDumper.Write(p);
+			}
+		}
+
+        [Category("Homework")]
+        [Title("Querie_1")]
+        [Description("This querie return all cutomers with revenue more than some x_sum.")]
+
+        public void Querie1()
```
Comment (hobowithshotgun) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: You should change methods' names to LinqXXX to make it work in program's UI

Please do this and I'll review

Comment (Avyalikov) [reply HW5/Vyalikov_Anton/Task/LinqSamples.cs]: Oh, such a stupid mistake, thanks.

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        [Category("Homework")]
+        [Title("Linq022")]
+        [Description("This querie return all providers from the same city or country for each customer (with group).")]
+
+        public void Linq022()
+        {
+            var suppliers = dataSource.Suppliers.GroupBy(a => new { a.City, a.Country });
+
+            var customers = dataSource.Customers.GroupBy(a => new { a.City, a.Country }).Select(gr => new
+            {gr.Key, Subgroups = gr.Select(a => new { a, Suppliers = suppliers.Where(b => b.Key.Equals(gr.Key)) })})
```
Comment (hobowithshotgun) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: This one outputs all order details that we don't need. Fix the select state please

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+
+        [Category("Homework")]
+        [Title("Linq006")]
+        [Description("This querie return all customers with non-num postcode or without region or without operator code.")]
+
+        public void Linq006()
+        {
+            var customers = dataSource.Customers
+                .Where(a => !a.PostalCode.All(char.IsDigit) || a.Region == null || !a.Phone.StartsWith("("))
+                .Select(a => new { a.CustomerID, a.CompanyName, a.PostalCode, a.Region, a.Phone});
```
Comment (hobowithshotgun) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: You have an ArgumentNullException here:
System.ArgumentNullException: Значение не может быть неопределенным.

Fix this please

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+        }
+
+        [Category("Homework")]
+        [Title("Linq006")]
+        [Description("This querie return all customers with non-num postcode or without region or without operator code.")]
+
+        public void Linq006()
+        {
+            var customers = dataSource.Customers
+                .Where(a => !SqlMethods.Like(a.PostalCode, "%[0-9]%") || a.Region == null || !a.Phone.StartsWith("("))
```
Comment (hobowithshotgun) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: `!SqlMethods.Like(a.PostalCode,` "%[0-9]%")` 
This one doesn't work and I dont know why. You should rewrite with ^: `SqlMethods.Like(a.PostalCode,`` "%[^0-9]%")`

Comment (Avyalikov) [reply HW5/Vyalikov_Anton/Task/LinqSamples.cs]: Hm, strange, but it worked for me.

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs (hunk 4 above)
Comment (Avyalikov) [reply HW5/Vyalikov_Anton/Task/LinqSamples.cs]: Yeah, my fault.

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs (hunk 3 above)
Comment (Avyalikov) [reply HW5/Vyalikov_Anton/Task/LinqSamples.cs]: It was a bit stupid and unreadable decision, I reworked it.

File: HW5/Vyalikov_Anton/Task/LinqSamples.cs [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+        }
+
+        [Category("Homework")]
+        [Title("Linq010")]
+        [Description("This querie return statistics of customers activity.")]
+
+        public void Linq010()
+        {
+            var ordersDate = dataSource.Customers.SelectMany(a => a.Orders.Select(b => b.OrderDate));
+            var monthStat = ordersDate.GroupBy(a => a.Month).Select(a => new { a.Key, MonthsOrders = a.Count() }).OrderBy(a => a.Key);
```
Comment (MilRoad) [inline HW5/Vyalikov_Anton/Task/LinqSamples.cs]: For the accurate statistic we need to divide months count by years count

**Pull request merged**